media_test_harness/testcase.py

```diff
@@ -37,24 +38,24 @@ def log_video_debug_lines(self):
             if debug_lines:
                 self.marionette.log('\n'.join(debug_lines))
 
-    def run_playback(self, interval=1, set_duration=None,
-                     stall_wait_time=10):
+    def run_playback(self, video):
         with self.marionette.using_context('content'):
-            for url in self.video_urls:
-                self.logger.info(url)
-                video = VP(self.marionette, url, interval=interval,
-                           set_duration=set_duration,
-                           stall_wait_time=stall_wait_time)
-                verbose_until(Wait(video, timeout=30), video,
-                              playback_started)
+            self.logger.info(video.test_url)
+            try:
+                verbose_until(Wait(video, interval=video.interval,
+                                   timeout=video.expected_duration * 1.3 +
```

Comment:
My reasons for the `1.3`, which is a loose upper bound:
* unpredictable ad behaviour makes it hard to come up with precise timeouts
* in the past I attributed many timeout-related failures to incredible slowness on pf-jenkins builders, but I can't get more specific than that because it was many months ago during initial development of the tests

I suggest adding a test that specifically looks for stalls in a strict manner, meanwhile keeping our basic playback tests looser. 

Aside: the way we measure lag depends on `current_time`, but when ads (or other spliced-in streams) are involved, `current_time` refers to whatever is playing right now (e.g. an ad), so `current_time` could suddenly be 0 in the middle of the main video, yielding a very large lag value.